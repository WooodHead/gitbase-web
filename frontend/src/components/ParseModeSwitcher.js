import React from 'react';
import PropTypes from 'prop-types';
import api from '../api';

function ParseModeSwitcher({ mode, handleModeChange }) {
  return (
    <div style={{ padding: '20px' }}>
      {api.uastModes.map(m => (
        <label
          key={m}
          style={{
            marginRight: '20px',
            textTransform: 'capitalize'
          }}
        >
          <input
            type="radio"
            value={m}
            checked={mode === m}
            onChange={e => handleModeChange(e.target.value)}
          />{' '}
          {m}
        </label>
      ))}
    </div>
  );
}

ParseModeSwitcher.propTypes = {
  mode: PropTypes.string.isRequired,
  handleModeChange: PropTypes.func.isRequired
};

export default ParseModeSwitcher;
