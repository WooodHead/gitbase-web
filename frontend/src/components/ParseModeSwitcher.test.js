import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ParseModeSwitcher from './ParseModeSwitcher';

test('it renders correctly', () => {
  const tree = renderer
    .create(
      <ParseModeSwitcher mode={'semantic'} handleModeChange={() => null} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('it calls handleModeChange when the input is clicked', () => {
  const spy = jest.fn();
  const wrapper = shallow(
    <ParseModeSwitcher mode={'semantic'} handleModeChange={spy} />
  );
  wrapper
    .find('[value="annotated"]')
    .simulate('change', { target: { value: 'annotated' } });
  expect(spy.mock.calls.length).toBe(1);
});
