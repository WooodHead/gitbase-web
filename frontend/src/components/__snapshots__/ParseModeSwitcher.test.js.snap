// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <label
    style={
      Object {
        "marginRight": "20px",
        "textTransform": "capitalize",
      }
    }
  >
    <input
      checked={true}
      onChange={[Function]}
      type="radio"
      value="semantic"
    />
     
    semantic
  </label>
  <label
    style={
      Object {
        "marginRight": "20px",
        "textTransform": "capitalize",
      }
    }
  >
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="annotated"
    />
     
    annotated
  </label>
  <label
    style={
      Object {
        "marginRight": "20px",
        "textTransform": "capitalize",
      }
    }
  >
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="native"
    />
     
    native
  </label>
</div>
`;
